protocols/imap/src/main/java/org/apache/james/imap/decode/parser/AppendCommandParser.java

```diff
@@ -47,39 +47,31 @@ public AppendCommandParser(StatusResponseFactory statusResponseFactory) {
      * "flag_list" argument from the request. If not, returns a MessageFlags
      * with no flags set.
      */
-    public Flags optionalAppendFlags(ImapRequestLineReader request) throws DecodingException {
+    private Flags parseFlags(ImapRequestLineReader request) throws DecodingException {
         char next = request.nextWordChar();
         if (next == '(') {
             return request.flagList();
-        } else {
-            return null;
         }
+        return new Flags();
     }
 
     /**
      * If the next character in the request is a '"', tries to read a DateTime
      * argument. If not, returns null.
      */
-    public LocalDateTime optionalDateTime(ImapRequestLineReader request) throws DecodingException {
+    private LocalDateTime parseDateTime(ImapRequestLineReader request) throws DecodingException {
         char next = request.nextWordChar();
         if (next == '"') {
             return request.dateTime();
-        } else {
-            return null;
         }
+        return LocalDateTime.now();
```

Comment:
I'm pretty sure this code is not tested because relying on `now` make it really hard.
Could you please ensure this is tested ?

Reply:
Maybe we can return an Optional instead of null previously, and move the call to now up the chain ? Or use a Clock ?

Reply:
I fear we never get done if we target always perfect code.

1 refactoring at a time?

Reply:
I've been contributing the use of clocks in a later commit FYI